mesonbuild/linkers.py

```diff
@@ -248,3 +263,359 @@ def unix_args_to_native(cls, args):
+        self.id = id
+        self.exe = exe
+        self.compiler_type = compiler_type
+
+    def can_linker_accept_rsp(self) -> bool:
+        """Determines whether the linker can accept arguments using the @rsp
+        syntax.
+        """
+        return mesonlib.is_windows()
+
+    def get_linker_always_args(self) -> List[str]:
+        """Get arguments always required by the linker."""
+        return []
+
+    def get_option_link_args(self, options) -> List[str]:
+        return []
+
+    def get_args_from_envvars(self) -> List[str]:
+        link_flags = os.environ.get('LDFLAGS', '')
+        if link_flags:
+            mlog.log('Appending {} from environment: {!r}'.format('LDFLAGS', link_flags))
+            return shlex.split(link_flags)
+        return []
+
+    def get_std_shared_lib_link_args(self) -> List[str]:
+        return []
+
+    def get_std_shared_module_link_args(self) -> List[str]:
+        return self.get_std_shared_lib_link_args()
+
+    def get_link_whole_for(self, args: List[str]) -> List[str]:
+        if isinstance(args, list) and not args:
+            return []
+        raise mesonlib.EnvironmentException('Language %s does not support linking whole archives.' % self.get_display_language())
+
+    def get_pie_link_args(self) -> List[str]:
+        """Arguments required by the dynamic linker for -fPIE executables."""
+        return []
+
+    def get_lto_link_args(self) -> List[str]:
+        """Arguments required by the dynamic linker for LTO/IPO."""
+        return []
+
+    def gen_export_dynamic_link_args(self, env) -> List[str]:
+        return []
+
+    def get_asneeded_args(self) -> List[str]:
+        return []
+
+    def get_buildtype_linker_args(self, buildtype: str) -> List[str]:
+        return self._buildtype_args[buildtype]
+
+    def get_linker_exelist(self) -> List[str]:
+        return self.exe.copy()
+
+    def get_allow_undefined_link_args(self) -> List[str]:
+        return []
+
+    def get_linker_lib_prefix(self) -> str:
+        return ''
+
+    def get_linker_search_args(self, dirname: str) -> List[str]:
+        return []
+
+    def get_linker_output_args(self, outputname: str) -> List[str]:
+        return []
+
+    def linker_to_compiler_args(self, args: List[str]) -> List[str]:
+        return args
+
+    def get_linker_debug_crt_args(self) -> List[str]:
+        """Arguments needed to select a debug crt for the linker.
+
+        This is only needed for MSVC
+
+        Sometimes we need to manually select the CRT (C runtime) to use with
+        MSVC. One example is when trying to link with static libraries since
+        MSVC won't auto-select a CRT for us in that case and will error out
+        asking us to select one.
+        """
+        # TODO: this really belongs in the static linker, no?
+        return []
+
+    def get_no_stdlib_link_args(self) -> List[str]:
+        return []
+
+    def get_linker_output_args(self, outputname: str) -> List[str]:
```

Comment:
__[Flake8]__ [F811] redefinition of unused 'get_linker_output_args' from line 346 ([link](https://sider.review/gh/repos/19784232/pulls/5148#issue-3915086))

<sub>posted by Sider</sub>